### PR TITLE
Add interface guidelines for Radio and RadioGroup

### PR DESCRIPTION
**Pull request by mperrotti**

### CODE REVIEW

**Review by keithamus — APPROVED** [review]
This LGTM :+1:

**Comment by mperrotti** [comment]
Don't merge this yet - I noticed a few mistakes I need to correct.

File: content/components/radio-group.mdx
Comment (emilybrick) [inline content/components/radio-group.mdx:3]:
```suggestion
description: Radio group is used to render a short list of mutually exclusive options.
```

very nitpicky suggestion based on new guidelines (that haven't merged yet) - Cole and I are trying to be more consistent with component descriptions (see [here](https://github.com/primer/design/blob/498f2016ed969204e954bce6c939076d3d72774e/content/guides/contribute/documentation.mdx#documentation-preferences))

**Comment by mperrotti** [comment]
ok, this can be merged as soon as I get an approval from @emilybrick

File: content/components/checkbox-group.mdx
Comment (emilybrick) [inline content/components/checkbox-group.mdx:35]:
```suggestion
**Label:** A title that labels the group of options and helps users understand the relationship between the options in the group.
```

File: content/components/checkbox-group.mdx
Comment (emilybrick) [inline content/components/checkbox-group.mdx:37]:
```suggestion
**Required indicator:** Indicates that a selection is required.
```

File: content/components/checkbox-group.mdx
Comment (emilybrick) [inline content/components/checkbox-group.mdx:39]:
```suggestion
**Caption:** May be used to provide additional context about the checkbox group when the label alone is not sufficient.
```

File: content/components/checkbox-group.mdx
Comment (emilybrick) [inline content/components/checkbox-group.mdx:41]:
```suggestion
**Options:** A list of options represented using [checkboxes](/components/checkbox).
```

**Review by emilybrick — APPROVED** [review]
✨

**Pull request merged**
